tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -2750,6 +2750,28 @@ def visit_import(self, node):
         except:
             pass
 
+class DoNotUseLegacyTyping(BaseChecker):
+
+    """ Rule to check that we aren't using legacy typing using comments. """
+
+    name = "do-not-use-legacy-typing"
+    priority = -1
+    msgs = {
+        "C4761": (
+            "Do not use legacy typing using comments.",
+            "do-not-use-legacy-typing",
+            "Do not use legacy typing using comments. Python 2 is no longer supported, use Python 3.9+ type hints instead.",
+        ),
+    }
+
+    def visit_functiontype(self, node):
```

Comment:
I wasn't able to find functiontype as a node, could you point to where you found it?

Reply:
I have linked it in my original comment, it is part of the Python `ast` package and is not present in the `astroid` package. Here is the link in full: [https://docs.python.org/3/library/ast.html#ast.FunctionType](https://docs.python.org/3/library/ast.html#ast.FunctionType)

Reply:
Thanks! Okay, the pylint custom checkers use the AST representation provided by the astroid library, so the visit_XXX functions available would be nodes defined by astroid: https://github.com/pylint-dev/astroid/blob/0156c045346b231d4451e8aea85128dcf4e56c64/astroid/nodes/as_string.py#L97. Just as an idea, an alternative here could be calling visit_functiondef and diving into the function [properties ](https://pylint.pycqa.org/projects/astroid/en/latest/api/nodes/astroid.nodes.FunctionDef.html)there.

Reply:
I see, that was a misunderstanding on my part. I've re-written the checker and tests, and re-run the checker against azure-sdk-for-python.

Reply:
Awesome thank you!